File: mobile/src/constants/analyticsEvents.ts
/**
 * Analytics event names and shared property types.
 *
 * Only events the mobile client itself emits belong here. The backend emits
 * every server-observable event on its own; the client posts only what the
 * backend cannot see (delivered via POST /api/analytics/events).
 */
export const ANALYTICS_EVENTS = {
  RECEIPT_ABANDONED: 'receipt_abandoned',
} as const;

export type AnalyticsEventName = (typeof ANALYTICS_EVENTS)[keyof typeof ANALYTICS_EVENTS];

/** Stage at which a receipt flow was abandoned (see docs/prd/12-analytics.md). */
export type AbandonStage = 'capture' | 'review';

/** Where a receipt scan originated. */
export type ScanSource = 'camera' | 'library' | 'pdf' | 'manual';

File: backend/app/services/analytics_events.py
"""Allowlist of analytics events the client may submit via the ingestion API.

Kept deliberately minimal: the backend already emits every server-observable
event itself, so a client only needs to deliver events the backend cannot see.
Restricting the allowlist prevents an authenticated user from injecting
server-owned events (e.g. ``budget_created``, ``receipt_confirmed``) and
skewing product funnels.
"""

# Events the client is permitted to POST to /api/analytics/events.
CLIENT_EMITTABLE_EVENTS: frozenset[str] = frozenset({"receipt_abandoned"})
